taichi/llvm/llvm_program.cpp

```diff
@@ -158,9 +158,9 @@ LlvmProgramImpl::clone_struct_compiler_initial_context(
   return tlctx->clone_runtime_module();
 }
 
-void LlvmProgramImpl::initialize_llvm_runtime_snodes(const SNodeTree *tree,
-                                                     StructCompiler *scomp,
-                                                     uint64 *result_buffer) {
+void LlvmProgramImpl::initialize_llvm_runtime_snodes(
+    const LlvmOfflineCache::FieldCacheData &field_cache_data,
```

Comment:
nit: I think the naming is a bit weird here, in the sense that init LLVM runtime SNodes shouldn't explicitly depend on some offline cache... Maybe we can extract this struct and give it a more generic naming in the future?

Reply:
Agreed. Let me rename FieldCacheData and make it a standalone struct. 

btw, does "RuntimeFieldData" or "CompiledFieldData" sound good to you? Or any alternative suggestions?

Reply:
Naming is hard... I don't have a good suggestion at this point, either. Maybe we should change this once SNodeMeta thing is more clear?

Reply:
Sure, so how about keep it as is for now? 

I'm planning to file another issue, talking about synchronizing the implementations for JIT, AOT and OfflineCache. Maybe this naming thing can be covered in that thread?